File: bench/python/sieve.py
# Sieve of Eratosthenes to N, counting primes. Checksum = count of primes <= N.
import os
N = int(os.environ.get("BENCH_N", "1000000"))
comp = bytearray(N + 1)
p = 2
while p * p <= N:
    if not comp[p]:
        for j in range(p * p, N + 1, p):
            comp[j] = 1
    p += 1
count = 0
for k in range(2, N + 1):
    if not comp[k]:
        count += 1
print(count)

File: bench/node/sieve.js
// Sieve of Eratosthenes to N, counting primes. Checksum = count of primes <= N.
const N = parseInt(process.env.BENCH_N || "1000000", 10);
const comp = new Uint8Array(N + 1);
for (let p = 2; p * p <= N; p++) {
  if (!comp[p]) for (let j = p * p; j <= N; j += p) comp[j] = 1;
}
let count = 0;
for (let k = 2; k <= N; k++) if (!comp[k]) count++;
console.log(count);
